### PR TITLE
TypeSpec changes for CSharp

### PR DESCRIPTION
**Pull request by wiboris**
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

File: specification/batch/Azure.Batch/client.tsp [hunk 1]
```diff
@@ -4,19 +4,26 @@ import "./main.tsp";
...
+@@access(Azure.Batch.PostOperation, Access.internal, "java");
+@@access(Azure.Batch.FileOperation, Access.internal, "java");
+@@access(Azure.Batch.HeadFileOperation, Access.internal, "java");
+@@access(Azure.Batch.ResourceInfoOperationWithAdditionalResponse,
   Access.internal,
   "java"
 );
 
+// Api overrides
+@@access(Azure.Batch.Batch.poolExists, Access.internal, "csharp");
```
Comment (dpwatrous) [inline specification/batch/Azure.Batch/client.tsp:22]: `Azure.Batch.Batch` looks strange - can we rename one of these?

File: specification/batch/Azure.Batch/tspconfig.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ options:
     package-dir: "Azure.Batch"
     clear-output-folder: true
     model-namespace: false
-    head-as-boolean: true
+    head-as-boolean: false
```
Comment (dpwatrous) [inline specification/batch/Azure.Batch/tspconfig.yaml:19]: What does this do for HEAD operations that are existence checks such as `poolExists`?

File: specification/batch/Azure.Batch/client.tsp (hunk 1 above)
Comment (dpwatrous) [reply specification/batch/Azure.Batch/client.tsp:22]: Maybe just rename the `Batch` interface to `BatchClient` so this is `@@access(Azure.Batch.BatchClient)`? This seems to match the pattern I see other teams using.

File: specification/batch/Azure.Batch/routes.tsp [hunk 3]
```diff
@@ -140,8 +23,11 @@ alias Created<T> = {
 #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
 #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Operation 'Exists' should be defined using a signature from the Azure.Core namespace."
 #suppress "@azure-tools/typespec-azure-core/rpc-operation-request-body" "BatchOperation with '@get' cannot have a body."
-@clientName("Batch")
-interface BatchApi {
+@client({
+  name: "BatchClient",
```
Comment (dpwatrous) [inline specification/batch/Azure.Batch/routes.tsp:27]: If we rename the interface here to `BatchClient` do we need the name property here? (Doesn't seem like it does any harm, but I wonder if it's redundant)

**Review by dpwatrous — CHANGES_REQUESTED** [review]

File: specification/batch/Azure.Batch/tspconfig.yaml (hunk 2 above)
Comment (wiboris) [reply specification/batch/Azure.Batch/tspconfig.yaml:19]: The issue I found is that If I make head-as-boolean true then poolExist will return bool so will getFileProperties.  I found it better for both to return a Response then customize them.

File: specification/batch/Azure.Batch/routes.tsp (hunk 3 above)
Comment (wiboris) [reply specification/batch/Azure.Batch/routes.tsp]: removed the name, doesn't appear to effect generation of the swagger or csharp sdk

File: specification/batch/Azure.Batch/client.tsp (hunk 1 above)
Comment (wiboris) [reply specification/batch/Azure.Batch/client.tsp]: changed it to BatchClient but it did have the side effect of changing the operationIds.

**Review by dpwatrous — APPROVED** [review]
After talking about the changes further, I'm ok if we do more refactoring in a later PR. Approved.

File: specification/batch/Azure.Batch/routes.tsp [hunk 4]
```diff
@@ -140,8 +23,11 @@ alias Created<T> = {
...
+  service: Azure.Batch,
+})
+interface Batch {
```
Comment (qiaozha) [inline specification/batch/Azure.Batch/routes.tsp:30]: rename client should be put into the client.tsp, we should never use any configuration from @azure-tools/typespec-client-generator-core within the scope of main.tsp and all the files that it refers to.

**Pull request merged**